kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/sessiontracker/SessionSerializer.java

```diff
@@ -425,6 +432,8 @@ private SessionInfo doSerialize(String sessionId,
         try (TransientInjectableObjectOutputStream outStream = TransientInjectableObjectOutputStream
                 .newInstance(out, handler, injectableFilter)) {
             outStream.writeWithTransients(attributes);
+        } catch (Exception ex) {
+            sessionSerializationCallback.onSerializationError(ex);
```

Comment:
I would throw the exception anyway after invoking the callback; otherwise we may end up writing an empty session if the user callback does not re-throw the exception.

Reply:
Makes sense. Added the exception throw.